### PR TITLE
builtin(list): add List.walkFrom, List.walkFromUntil

### PR DESCRIPTION
**Pull request by snprajwal**
Implements two more List funcs listed in #664 :)

### CODE REVIEW

File: crates/compiler/test_gen/src/gen_list.rs [hunk 1]
```diff
@@ -3520,6 +3516,16 @@ fn list_walk_backwards_implements_position() {
...
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm"))]
+fn list_walk_backwards_until_sum() {
+    assert_evals_to!(
+        r#"List.walkBackwardsUntil [1, 2] 0 \a,b -> Continue (a + b)"#,
+        3,
+        i64
+    );
+}
+
```
Comment (ayazhafiz) [inline crates/compiler/test_gen/src/gen_list.rs:3528]: Not a big deal, but is there a reason to reorder the test here?

Comment (snprajwal) [reply crates/compiler/test_gen/src/gen_list.rs:3528]: One of the other merges separated this test from another test of the same func, so I just moved them one below another

**Comment by ayazhafiz** [comment]
You may need to rebase on main to pull in the recent fixes to the markdown links checks

**Review by ayazhafiz — APPROVED** [review]
LGTM, thank you!

**Pull request merged**